src/sw-utils.js

```diff
@@ -13,32 +13,74 @@
  * @function injectServiceWorker
  * @param {funtion} updateCallback - Callback to be triggered when service worker update is found.
  */
-function injectServiceWorker(updateCallback) {
+
+const createFrame = (id, innerHTML, classList) => {
+    const frame = document.createElement('div');
+    frame.setAttribute('id', id);
+    frame.innerHTML = innerHTML;
+    frame.classList.add(classList);
+    document.body.appendChild(frame);
+};
+
+const hideFrame = id => document.getElementById(id).classList.add('hidden-frame');
+
+const showFrame = id => document.getElementById(id).classList.remove('hidden-frame');
+
+function updateExperience(sw) {
+    let refreshing;
+    document.getElementById('ignore-update').addEventListener('click', () => hideFrame('app-update-frame'));
+    navigator.serviceWorker.addEventListener(
+        'controllerchange',
+        () => {
+            if (refreshing) return;
+            refreshing = true;
+            window.location.reload();
+        },
+    );
+    showFrame('app-update-frame');
+    document.getElementById('update-app').addEventListener('click', () => {
+        hideFrame('app-update-frame');
+        showFrame('in-flight-requests-frame');
+        sw.postMessage({
+            type: 'SKIP-WAITING',
+        });
+    });
+}
+
+function injectServiceWorker({ withUpdate = false, cb } = {}) {
     if ('serviceWorker' in navigator) {
         navigator.serviceWorker.register('/service-worker.js')
             .then((reg) => {
+                if (withUpdate) {
+                    const appUpdateTemplate = "<div id='update-txt'><img src='https://retail.flixcart.com/www/fk-retail-vpp/icon-star.png' /><span id='message'>Catch up with the latest version.</span><a id='update-app'>Update</a></span><a id='ignore-update'>X</a></div>";
```

Comment:
why not separate out these constants to separate file will make this function cleaner